### ISSUE TITLE
feat: add fenced-code-language rule (MD040)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#88]
## Overview

Implement the `fenced-code-language` rule.

Fenced code blocks must specify a language identifier. This is one of the most commonly violated Markdown rules and helps ensure proper syntax highlighting in renderers.

Related: #76 (Priority 1 — #1)

## Rule Behavior

**Fail:**
````markdown
```
some code here
```
````

**Pass:**
````markdown
```go
some code here
```
````

## Equivalents

| Tool | Rule |
|---|---|
| markdownlint | MD040 `fenced-code-flag` |
| remark-lint | `fenced-code-flag` |

## Implementation Notes

- Detect fenced code blocks (` ``` ` or `~~~`) that have no language identifier on the opening fence
- Should be togglable via config
- Must respect frontmatter (already stripped before linting)

## Acceptance Criteria

- [ ] Rule implementation
- [ ] Unit tests
- [ ] Config flag to enable/disable
- [ ] Documentation update
